waiter/src/waiter/auth/jwt.clj

```diff
@@ -133,24 +135,88 @@
   (get-key-id->jwk [this]
     "Returns a map for public key id to the JSON Web Key which contains the public key used to
      verify any JSON Web Token (JWT) issued by the authorization server.")
+  (request-access-token [this request oidc-callback-uri access-code code-verifier]
+    "Returns an async channel that will return the access token for the provided access code.")
+  (retrieve-authorize-url [this request oidc-callback-uri code-verifier state-code]
+    "Returns the OIDC authorize url to use for a given request.")
   (retrieve-server-state [this include-flags]
     "Returns the current state of the auth server."))
 
-(defrecord JwtAuthServer [jwks-url keys-cache]
+(defrecord JwtAuthServer [http-client jwks-url keys-cache oidc-authorize-uri oidc-token-uri]
   AuthServer
   (get-key-id->jwk [_]
     (get @keys-cache :key-id->jwk))
+
+  (request-access-token [_ request oidc-callback-uri access-code code-verifier]
```

Comment:
Let's add some metrics around this.  
1. Time taken to retrieve access token
2. Number of pending requests.

Reply:
Added requested metrics.